head.html

```diff
@@ -20,3 +20,6 @@
 <link rel="apple-touch-icon" sizes="167x167" href="/icons/apple-touch-icon-167x167.png"/>
 <link rel="apple-touch-icon" sizes="180x180" href="/icons/apple-touch-icon-180x180.png"/>
 <link rel="apple-touch-icon-precomposed" href="/icons/apple-touch-icon-precomposed.png"/>
+<script>
+    if (!navigator.sendBeacon) { var data = JSON.stringify({ referer: window.location.href, checkpoint: 'unsupported', weight: 1 }); var img = new Image(); img.src = 'https://rum.hlx3.page/.rum/1?data='+encodeURIComponent(data); }
```

Comment:
Two thoughts:

1. checking for `!navigator.sendBeacon && !window.fetch` instead would only give us 1% more coverage, probably not worth the polyfill
2. we can get the `id` property back using


```javascript
var hashCode = function hashCode(s) {
  return s.split('').reduce(function (a, b) {
    return (a << 5) - a + b.charCodeAt(0) | 0;
  }, 0);
};

var id = "".concat(hashCode(window.location.href), "-").concat(new Date().getTime(), "-").concat(Math.random().toString(16).substr(2, 14));
```

(I took https://github.com/adobe/theblog/blob/1a171a0c48705469b7945a41af86c85ba357911b/scripts/v2/scripts.js#L28-L29 and ran it through `babel` with an IE11 preset) for a complete change:

```suggestion
    if (!navigator.sendBeacon) { 
    var hashCode = function hashCode(s) {
      return s.split('').reduce(function (a, b) {
        return (a << 5) - a + b.charCodeAt(0) | 0;
      }, 0);
    };

    var id = "".concat(hashCode(window.location.href), "-").concat(new Date().getTime(), "-").concat(Math.random().toString(16).substr(2, 14));

    var data = JSON.stringify({ referer: window.location.href, checkpoint: 'unsupported', weight: 1, id: id }); var img = new Image(); img.src = 'https://rum.hlx3.page/.rum/1?data='+encodeURIComponent(data); }
```

Reply:
the problem is that this would further pollute the `head.html` at this point, and i am not sure that this is worth it.

Reply:
do you think we could auto create the id on the RUM edge, if it is not there?
in this particular case it is pretty much guaranteed that one wouldn't get other checkpoint beacons from the same "page view"